doc/developers/advanced_installation.rst

```diff
@@ -22,127 +33,191 @@ basis to help users test bleeding edge features or bug fixes::
 Building from source
 =====================
 
-In the vast majority of cases, building scikit-learn for development purposes
-can be done with::
+.. _git_repo:
 
-    pip install cython pytest flake8
+Use `Git <https://git-scm.com/>`_ to check out the latest source from the
+`scikit-learn repository <https://github.com/scikit-learn/scikit-learn>`_ on
+`Github <https://github.com>`_ and pip to build the project::
 
-Then, in the main repository::
+    git clone git://github.com/scikit-learn/scikit-learn.git
+    cd scikit-learn
 
+    pip install cython
     pip install --editable .
 
-Please read below for details and more advanced instructions.
+Check that the installed scikit-learn with a version number ending with
+`.dev0`::
+
+    python -c "import sklearn; sklearn.show_versions()"
+
+Run the tests with ``pytest``::
+
+    pip install pytest
+    pytest sklearn
+
+Please read below for more details and :ref:`platform system specific
+instructions <platform_specific_instructions>` (e.g. in case compilation error
+message).
+
+.. note::
+
+    You will have to re-run the ``pip install --editable .`` command every
```

Comment:
Maybe we can keep it simple and say 

> You will have to re-run the ``pip install --editable .`` command every time you modify a Cython file (ending in `.pyx` or `.pxd`) to re-compile the extensions.

The rest (branch switching) becomes implicit but hopefully obvious for someone who would ever touch Cython code.